plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveExecutionInterceptor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.glue;
+
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.services.glue.model.UpdateTableRequest;
+
+public class GlueHiveExecutionInterceptor
+        implements ExecutionInterceptor
+{
+    private final boolean skipArchive;
+
+    GlueHiveExecutionInterceptor(boolean isSkipArchive)
+    {
+        this.skipArchive = isSkipArchive;
+    }
+
+    @Override
+    public SdkRequest modifyRequest(Context.ModifyRequest context, ExecutionAttributes executionAttributes)
+    {
+        if (context.request() instanceof UpdateTableRequest updateTableRequest) {
```

Comment:
I believe handling only `UpdateTableRequest` is insufficient. For instance, `BatchCreatePartitionRequest` is used when inserting into partitioned tables.

Reply:
The new question is why we don't list possible operations unlike `SkipArchiveRequestHandler`.

Reply:
If we do that, this would make sure don't miss any operations that would benefit from this flag. But this assumes we have test coverage for all operations. I only added a test for CREATE TABLE and INSERT, should I add more DML queries?

Reply:
Let's leave it as-is.